docs/data/date-pickers/custom-components/PopperComponent.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import dayjs from 'dayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+
+export default function PopperComponent() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DemoContainer components={['DatePicker']}>
+        <DatePicker
+          label="Basic date picker"
+          views={['month', 'year']}
+          minDate={dayjs(new Date(2022, 0, 1))}
+          maxDate={dayjs(new Date(2024, 12, 31))}
+          reduceAnimations
+          slotProps={{
+            popper: {
+              modifiers: [
```

Comment:
Going with the modifier felt cleaner to me. If it were just the height that needed to be changed, I would have suggested maybe going with a custom popper in the slots, or applying the styling in a different way. However, if we make the popper have a fixed height, and one of the views is considerably smaller than the others, the smaller view shifts within the popper (depending on positioning), rather than sticking close to the input. I felt like the modifier exposed a nice way to control how things get displayed based on positioning. Does that make sense? Maybe you have other suggestions? 🤔

Reply:
Gotcha. Thanks for clarifying.
No, I don't have any better solution suggestions and the existing one is great, it serves the purpose well. 👍 
One additional note is that we might also consider stressing the fact that this change might be relevant to users with a very limited selection of allowed years.